system/jlib/jthread.cpp

```diff
@@ -547,10 +547,26 @@ CThreadedPersistent::CThreadedPersistent(const char *name, IThreaded *_owner) :
 
 CThreadedPersistent::~CThreadedPersistent()
 {
-    join(INFINITE);
+    try
+    {
+        join(INFINITE);
```

Comment:
Better would be a parameter on join() to prevent throwing an exception - otherwise if this is cleaning up as a result of an exception being thrown it will terminate anyway.